tests/app/main/views/test_xss.py

```diff
@@ -0,0 +1,35 @@
+# coding=utf-8
+from __future__ import unicode_literals
+
+import mock
+from ...helpers import LoggedInApplicationTest
+from dmapiclient import HTTPError
+import urllib
+
+
+class TestXSS(LoggedInApplicationTest):
+    @mock.patch('app.main.views.services.data_api_client')
+    def test_service_not_found_flash_message_injection(self, data_api_client):
+        """
+        Asserts that raw HTML in a bad service ID cannot be injected into a flash message.
+        """
+        # impl copied from test_services.TestServiceView#test_redirect_with_flash_for_api_client_404
+        api_response = mock.Mock()
+        api_response.status_code = 404
+        data_api_client.get_service.side_effect = HTTPError(api_response)
+
+        evil_service_id = "1<img src=a onerror=alert(1)>"
+        response1 = self.client.get('/admin/services/' + urllib.quote(evil_service_id))
```

Comment:
I'd put the literal escaped text here instead of using `urllib.quote`. This solves the Py3 failures, and I'd argue that it makes the test clearer in a way, since you can see exactly what's being sent.